tests/basic/reboot_test.go

```diff
@@ -253,3 +253,93 @@ var _ = Describe("{ReallocateSharedMount}", func() {
 		AfterEachTest(contexts, testrailID, runID)
 	})
 })
+
+var _ = Describe("{ClusterPxRestart}", func() {
+	var testrailID = 60042
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/60042&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=4120
+	var runID int
+	loc, _ := time.LoadLocation("UTC")
+	t1 := time.Now().In(loc)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("ClusterPxRestart", "Validate restart of PX in cluster", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+
+	var contexts []*scheduler.Context
+
+	stepLog := "has to restart cluster with volumes"
+	It(stepLog, func() {
+		dash.Info(stepLog)
+		//var err error
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("rebootPX-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+		stepLog = "get all nodes and reboot PX one by one"
+		Step(stepLog, func() {
+			dash.Info(stepLog)
+
+			Step("get all PX nodes and reboot one by one", func() {
+				nodesToReboot := node.GetWorkerNodes()
+
+				// Reboot node and check driver status
+				Step(fmt.Sprintf("reboot px one at a time from the node(s): %v", nodesToReboot), func() {
+					for _, n := range nodesToReboot {
+						if n.IsStorageDriverInstalled {
+							Step(fmt.Sprintf("reboot node: %s", n.Name), func() {
+
+								err := Inst().V.RestartDriver(n, nil)
+								dash.VerifyFatal(err, nil, "Restart PX")
+							})
+							Step(fmt.Sprintf("wait for node: %s to be back up", n.Name), func() {
+								err := Inst().N.TestConnection(n, node.ConnectionOpts{
+									Timeout:         defaultTestConnectionTimeout,
+									TimeBeforeRetry: defaultWaitRebootRetry,
+								})
+								dash.VerifyFatal(err, nil, "Validate node is back up")
+							})
+
+							Step(fmt.Sprintf("wait for volume driver to stop on node: %v", n.Name), func() {
+								err := Inst().V.WaitDriverDownOnNode(n)
+								dash.VerifyFatal(err, nil, "Validate node is back up")
+							})
+
+							Step(fmt.Sprintf("wait to scheduler: %s and volume driver: %s to start",
+								Inst().S.String(), Inst().V.String()), func() {
+
+								err := Inst().S.IsNodeReady(n)
+								dash.VerifyFatal(err, nil, "Validate PX Node is back up")
+
+								err = Inst().V.WaitDriverUpOnNode(n, Inst().DriverStartTimeout)
+								dash.VerifyFatal(err, nil, "Validate PX driver is back up")
+							})
+
+							Step("validate apps", func() {
+								for _, ctx := range contexts {
+									ValidateContext(ctx)
+								}
+							})
+						}
+					}
+				})
+			})
+		})
+		t2 := time.Now()
+		fmt.Printf("Time taken to reboot all the PX is: %f", t2.Sub(t1).Hours())
```

Comment:
use log or dash.
also u can add averfircation for the time , define expected tine based on number of nodes